server.go

```diff
@@ -69,6 +79,14 @@ type Server struct {
 
 // ServerConfig defines the configuration for the Server
 type ServerConfig struct {
+	// StateChanges receives state transition details each time client
+	// connection state changes. The channel is expected to be sufficiently
+	// buffered to keep up with event pace.
+	//
+	// If nil, no information about state transitions are dispatched
+	// by the library.
+	StateChanges chan<- *ClientStateChange
```

Comment:
Can I ask you one more thing? If we're going to use the same ClientStateChange for both server and client, how about renaming it to just `StateChange`?